File: Java/Software Development Module - Java/Revision/Exam2018/InterestCalculatorApp.java
/*
InterestCalculatorApp.java
Billy Cussen
03/09/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class InterestCalculatorApp{

    public static void main(String [] args){

        //Variables
        boolean inputCheck;
        boolean accessMoney = true;
        double amount = 0.0; 
        double[] allInterest;
        int allInterestLength = 0;
        int months = 0;
        String retry = new String();

        //Objects
        InterestCalculator myInterestCalculator = new InterestCalculator();
        Scanner keyboard;
        
        do{
            inputCheck = true;
            keyboard = new Scanner(System.in);
            try{
                System.out.print("Please enter the number of Deposits to be Checked: ");
                allInterestLength = keyboard.nextInt();
                if(allInterestLength <= 0 || allInterestLength > 10){
                    throw new InputMismatchException();
                }
            }catch(InputMismatchException e){
                System.out.println("You can check between 1 and 10 Deposits at a Time");
                inputCheck = false;
            }finally{
                keyboard = null;
            }

        }while(!inputCheck);

        allInterest = new double[allInterestLength];

        for(int i = 0; i < allInterest.length; i++){

            //Input
            do{
                inputCheck = true;
                keyboard = new Scanner(System.in);
                try{
                    System.out.print("Please enter amount to be Deposited (Minimum is 1000/Max is 100000): ");
                    amount = keyboard.nextDouble();
                    if(amount < 1000 || amount > 100000){
                        throw new InputMismatchException();
                    }
                }catch(InputMismatchException e){
                    System.out.println("You need to enter a number to Note the amount to be Deposit");
                    inputCheck = false;
                }finally{
                    keyboard = null;
                }
            }while(!inputCheck);

            do{
                inputCheck = true;
                keyboard = new Scanner(System.in);
                try{
                    System.out.print("Please enter Months for Fixed Term Account Type (6,12 or 18): ");
                    months = keyboard.nextInt();
                    if(!(months == 6 || months == 12 || months == 18)){
                        throw new InputMismatchException();
                    }
                }catch(InputMismatchException | NumberFormatException e){
                    System.out.println("You need to enter 6, 8 or 12.");
                    inputCheck = false;
                } finally{
                    keyboard = null;
                }
            }while(!inputCheck);

            do{
                String answer = new String();
                inputCheck = true;
                keyboard = new Scanner(System.in);
                try{
                    System.out.print("Please enter Months for Fixed Term Account Type (\"Y\" or \"N\"): ");
                    answer = keyboard.nextLine();
                    if(!(answer.equalsIgnoreCase("Y")||answer.equalsIgnoreCase("N"))){
                        throw new InputMismatchException();
                    }

                    if(answer.equalsIgnoreCase("Y")){
                        accessMoney = true;
                    }else{
                        accessMoney = false;
                    }

                }catch(InputMismatchException | NumberFormatException e){
                    System.out.println("You need to enter \"Y\" or \"N\"");
                    inputCheck = false;
                } finally{
                    keyboard = null;
                }
            }while(!inputCheck);

            //Process & Output
            myInterestCalculator = new InterestCalculator();
            
            try{
                System.out.println("Total Interest based on the above is: "+myInterestCalculator.getInterest(amount,months,accessMoney));
            }catch(InputMismatchException e){
                System.out.println(e);
            } finally{
                allInterest[i] = myInterestCalculator.getInterest(amount,months,accessMoney);
            }
        }

        System.out.println("The Highest Interest to be paid is: "+myInterestCalculator.getHighestInterest(allInterest));
        myInterestCalculator = null;

    }

}

File: Python/Software Development Module - Python/Revision/Exam2018/InterestCalculator/InterestCalculatorApp.py
"""
InterestCalculatorApp.py
Billy Cussen
02/11/2020
"""

#Objects
import gc
import InterestCalculator
from InterestCalculator import myInterestCalculator

#Variable
repeat = "y"

while repeat == "y":
    inputCheck = False
    allDeposits = []
    myInterestCalculatorApp = myInterestCalculator()

    #Input - Number of Deposits!
    while inputCheck == False:
        try:
            numberOfDeposits = int(input("Please enter the number of Deposits to be checked (10 Max please!): "))
            if numberOfDeposits <= 0 or numberOfDeposits > 10:
                raise Exception
        except Exception:
            print("You need to enter an Integer between 1 and 10(e.g. 1,2,3...).")
        else:
            inputCheck = True
    
    for d in range(numberOfDeposits):
        inputCheck = True
        #Input - Amount to be Deposited
        while inputCheck:
            try:
                amount = float(input("Please enter the amount to be Deposited for Deposit "+str(d+1)+" (Min: 1000, Max: 10000): "))
                if amount < 1000 or amount > 10000:
                    raise Exception
            except Exception:
                print("Minimum that can be deposited is 1000, Maximum is 10000.")
            else:
                inputCheck = False

        #Input - Length of Term
        while inputCheck == False:
            try:
                months = int(input("Please enter the months for the Fixed Term Account (6, 12 or 18) for Deposit "+str(d+1)+": "))
                if months != 6 and months != 12 and months != 18:
                    raise Exception
            except Exception:
                print("You need to enter either 6, 12 or 18")
            else:
                inputCheck = True

        #Input - Can Money be Accessed
        while inputCheck:
            try:
                answer = str(input("Please enter if customer will be able to access money during this time (\"Y\" or \"N\"): "))
                answer = answer.lower()
                if answer != "y" and answer != "n":
                    raise Exception
            except Exception:
                print("You need to enter either \"Y\" or \"N\"")
            else:
                if answer == "y":
                    accessMoney = True
                else:
                    accessMoney = False
                inputCheck = False
        
        #Process & Output
        try:
            interest = myInterestCalculatorApp.getInterest(amount, months, accessMoney)
            result = "Total Interest for Customer "+str(d+1)+" is: "+str(interest)
        except Exception:
            print("An Error Occured when calculating interest for customer "+str(d+1))
        else:
            print(result)
            allDeposits.append(interest)

    try:
        highestInterestAmount = myInterestCalculatorApp.getHighestInterest(allDeposits)
        highestInterestResult = "Highest Interest for this batch of Deposits is: "+str(highestInterestAmount)
    except Exception:
        print("An Error Occured calculating Highest Interest of Deposits input")
    else:
        print(highestInterestResult)
        myInterestCalculatorApp = None
        allDeposits = None
        gc.collect()

    repeat = ""
    while repeat != "y" and repeat !="n":
        repeat = str(input("Please enter \"y\" to continue and \"n\" to exit: "))
        if repeat != "y" and repeat !="n":
            print("You need to enter \"y\" or \"n\"!")

print("Thanks for using my Interest Calculator App!")